tests/queries/0_stateless/03350_alter_table_fetch_partition_thread_pool.sql

```diff
@@ -0,0 +1,19 @@
+-- Tags: no-parallel, no-replicated-database
+-- Tag: no-parallel - to avoid polluting FETCH PARTITION thread pool with other fetches
+-- Tag: no-database-replicated - replica_path is different
+
+drop table if exists data1;
+drop table if exists data2;
+
+create table data1 (key Int) engine=ReplicatedMergeTree('/tables/{database}/{table}', 'r1') order by ();
+create table data2 (key Int) engine=ReplicatedMergeTree('/tables/{database}/{table}', 'r1') order by ();
```

Comment:
I see the test passed, but I am not sure why giving the same replica name works. Could you please explain it?

Reply:
It does not need any replicas at all note a `{table}` macro

It just need ReplicatedMergeTree. You can think about data1 as data table on server1, and data2 as a separate table (not a replica of data1) on server2, and then I'm using `FETCH PARTITION` to copy content from one table to another.

Reply:
Sorry, I completely missed the important part.